### ISSUE TITLE
RangeError findByIdUpdate call with nested sub document causes Maximum call stack size exceeded

### ISSUE THREAD

**Issue by monk8800** [Automattic/mongoose#4466]
Hi guys am noticing some strange behavior when trying to make an update to a document with nested sub document having a field type of string array. calling findByIdUpdate seems to cause “Maximum call stack size exceeded”. This issue seems to only happen on linux/Mac environments while windows seems to fine.  

Am using the following versions  

Node : **4.4.7**

mongoose : **4.5.9**

```
var EventTriggerScheduleSchema = new mongoose.Schema({
    interval: {
        type: [String], //causes error infinite loop 
        required: true
    }
});

var TriggerCoreSchema = new mongoose.Schema({
    name: {
        type: String,
        required: 'Name is required',
        minLength: 3,
        maxLength: 255,
        trim: true
    },
    schedule: EventTriggerScheduleSchema,
}, {
    timestamps: true
});

var id = "ObjectIdHere"; 
var updateObj = {
    "name": "MyEvent",
    "schedule": {
        "interval": [
            "MONDAY",
            "TUESDAY",
            "FRIDAY"
        ]
    },
    "__v": 0
};

TriggerCore.findByIdAndUpdate(id, updateObj, { new: true },
    function(err, updatedTrigger) {
        if (err) return cb(err);
        cb(null, updatedTrigger);
    });
```

Origin of error recursive call in **/lib/services/common.js** call to function **modifiedPaths(update, path, result)** seems to cause an infinite loop for type string array

**Comment by vkarpov15** [comment]
Can you please provide a code sample? "make an update to a document with nested sub document having a field type of string array" can mean a lot of different things in mongoose. Also, can you clarify which version of mongoose you're using?

**Comment by monk8800** [comment]
@vkarpov15, I have updated the original question. let me know if its not clear I will try to elobrate more. Thanks

**Comment by cfirestone** [comment]
I've been battling with this one too. I was on 4.5.10 of mongoose, but then rolled back to 4.4.20 and am no longer experiencing the issue.

**Comment by monk8800** [comment]
@cfirestone I ended up rewritting my function replaced findByIdAndUpdate function call with findOne followed by calling save on the found mongoose object.

**Comment by sgsfatcu** [comment]
Also encountered this issue after upgrading from mongoose 4.4.6 to 4.5.10.

In my case I was using the Model.update() function, and the array was of another schema.

**Comment by vkarpov15** [comment]
Looks like a duplicate of #4465 , should be fixed when 4.6.1 gets released
